packages/core/src/application.ts

```diff
@@ -61,9 +62,70 @@ export class Application extends Context {
    */
   controller(controllerCtor: ControllerClass, name?: string): Binding {
     name = name || controllerCtor.name;
-    return this.bind(`controllers.${name}`)
+    return this.bind(`${CoreBindings.CONTROLLERS_PREFIX}.${name}`)
       .toClass(controllerCtor)
-      .tag('controller');
+      .tag(CoreBindings.CONTROLLERS_TAG);
+  }
+
+  /**
+   * Register a booter class / array of classes with this application.
+   *
+   * @param {Function | Function[]} booterCls The booter class (constructor function).
+   * @param {string=} name Optional booter name, defaults to the class name.
+   * Ignored is cls is an Array and the name defaults to the class name.
+   * @returns {Binding | Binding[]} The newly created binding(s), you can use the
+   * reference to further modify the binding, e.g. lock the value to prevent
+   * further modifications.
+   *
+   * ```ts
+   * class MyBooter implements Booter {}
+   * app.booter(MyBooter);
+   * ```
+   */
+  booter(booterCls: Constructor<Booter>, name?: string): Binding;
+  booter(booterCls: Constructor<Booter>[]): Binding[];
+  booter(
+    booterCls: Constructor<Booter> | Constructor<Booter>[],
+    name?: string,
+    // tslint:disable-next-line:no-any
+  ): any {
+    if (Array.isArray(booterCls)) {
+      return booterCls.map(cls => this._bindBooter(cls));
+    } else {
+      return this._bindBooter(booterCls, name);
+    }
+  }
+
+  /**
+   *
+   * @param booterCls A Booter Class
+   * @param {string} name Name the Booter Class should be bound to
+   * @returns {Binding} The newly created Binding
+   */
+  private _bindBooter<T extends Booter>(
+    booterCls: Constructor<T>,
+    name?: string,
+  ): Binding {
+    name = name || booterCls.name;
+    return this.bind(`${CoreBindings.BOOTER_PREFIX}.${name}`)
+      .toClass(booterCls)
+      .inScope(BindingScope.CONTEXT)
+      .tag(CoreBindings.BOOTER_TAG);
+  }
+
+  /**
+   * Function is responsible for calling all registered Booter classes that
+   * are bound to the Application instance. Each phase of an instance must
+   * complete before the next phase is started.
+   * @param {BootOptions} bootOptions Options to use to boot the Application
+   */
+  async boot(bootOptions: BootOptions): Promise<void> {
+    try {
+      const bootstrapper = await this.get(CoreBindings.BOOTSTRAPPER);
+      await bootstrapper.boot(bootOptions);
+    } catch (err) {
```

Comment:
Please consider using the recently landed feature adding `optional` flag to `this.get` - see https://github.com/strongloop/loopback-next/pull/946.

```ts
async boot(bootOptions: BootOptions): Promise<void> {
  const bootstrapper = await this.get(CoreBindings.BOOTSTRAPPER, {optional: true});
  if (bootstrapper) {
    await bootstrapper.boot(bootOptions);
  } else {
    console.warn(`No bootstrapper was bound to ${CoreBindings.BOOTSTRAPPER}`);
  }
}
```

If you prefer to keep using try/catch, then please ensure that errors not related to missing BOOTSTRAPPER bindings are still correctly reported to the user. For example, when the bootstrapper is bound, but some of its dependencies cannot be resolved for whatever reasons (e.g. `MyArtifactBooter` depends on `foo.bar` which was not bound), then the warning `No bootstrapper was bound` is totally misleading.

Similarly, `bootstrapper.boot` must be outside of try/catch block.

Consider adding tests to verify that `app.boot` (re)throws errors caused by bootstrapper dependencies or `bootstrapper.boot`.